Stork/Stork/Source/JsonValue.swift

```diff
@@ -70,6 +76,16 @@ public enum JsonValue {
     }
   }
 
+  public func ifArray<T>(_ apply: ([JsonValue]) throws -> T?) -> T? {
+    switch self {
+    case .array(let jsonValue):
+      do { return try apply(jsonValue) }
+      catch { return nil }
+    default:
+      return nil
+    }
+  }
+
```

Comment:
The same goes about this one. This would only be useful to define that a type `T: FromJson` can be built from a JSON array.

For example, consider a simple `ReadingList` type:

``` swift
struct ReadingList {
  let books: [Book]
}
```

This `ifArray` method would be useful like so:

``` swift
public static func from(value: JsonValue) -> ReadingList? {
  return value.ifArray { array in
    ReadingList(books: array.compactMap { Book.from(value: $0)  })
  }
}
```

**Now**, note that this represents much more work than the following alternatives:

``` swift
// At a high-level
let readingList = ReadingList(
  books: [Book].from(jsonArray: booksArray)
)

// Or when parsing a field from an object
readingList: ReadingList(books: try json ..! "books")
```


### Alternatives

Despite these arguments, I do like the idea to support arrays as Stork does for all other json value types.

#### Alternative 1
A way of having them in a useful, more convenient way would be to add a convenient way of building an `[T]` from `[JsonValue]` so that the `ifArray` scenario becomes:

``` swift
public static func from(value: JsonValue) -> ReadingList? {
  return value.ifArray { array in
    ReadingList(books: [Book].from(jsonArray: array))
  }
}
```

Or a way of inferring the type desired to be obtained from `[JsonValue]`, so to have:

``` swift
public static func from(value: JsonValue) -> ReadingList? {
  return value.ifArray { array in
    ReadingList(books: array.parsed())
  }
}
```

Or even better:

``` swift
public static func from(value: JsonValue) -> ReadingList? {
  return ReadingList(books: value.parsedArray())
}
```

#### Alternative 2 


Introduce a `FromJsonArray<T>` protocol, so that the final type can be immediately inferred and having Stork only requiring types that are built from JSON arrays to do:

``` swift
extension ReadingList: FromJsonArray<Book>
```

Although it looks nice, it'd introduce incongruity to the simple Stork design of `FromJson`, as now the parsers would need to handle the `FromJson` and `FromJsonArray<T>` cases.


### Conclusion

I am open for discussion. I like the idea of supporting fully arrays for a matter of consistency but I don't see the objective benefit of doing so.

If we introduce this feature, I'd prefer to neat pick the simpler solutions proposed in the `Alternative 1`.